pkg/arvo/sys/vane/ames.hoon

```diff
@@ -2386,27 +2386,43 @@
         ?:  =(0 (end 0 bone))
           ~
         `u=message-pump-state
-      ::  clogged: are five or more response messages unsent to this peer?
       ::
       =/  clogged=?
-        =|  acc=@ud
-        |-  ^-  ?
-        ?~  pumps
-          %.n
-        =.  acc
-          %+  add  acc
-          %+  add
-            ::  in-flight messages
+        |^  &(nuf-messages nuf-memory)
+        ::  +nuf-messages: are there enough messages to mark as clogged?
+        ::
+        ++  nuf-messages
+          =|  num=@ud
+          |-  ^-  ?
+          ?~  pumps  |
+          =.  num
+            ;:  add  num
+              (sub [next current]:i.pumps)
+              ~(wyt in unsent-messages.i.pumps)
+            ==
+          ?:  (gte num 5)
+            &
+          $(pumps t.pumps)
+        ::  +nuf-memory: is enough memory used to mark as clogged?
+        ::
+        ++  nuf-memory
+          =|  mem=@ud
+          |-  ^-  ?
+          ?~  pumps  |
+          =.  mem
+            ;:  add  mem
+              %+  roll  ~(tap to unsent-messages.i.pumps)
+              |=([a=@ b=@ud] (add b (met 3 a)))
```

Comment:
This could be further golfed for efficiency:

```
%+  add
  %-  ~(rep in unsent-messages.i.pumps)
  |=([a=@ b=_mem] (add b (met 3 a)))
?~  unsent-fragments.i.pumps. 0
(met 3 fragment.i.unsent-fragments.i.pumps)
```

that saves the list allocation (`+tap:to`) and one `+add` call (by initializing the `+rep:in` accumulator to `mem`). Using `+rep:in` on a `+qeu` looks strange, but works fine.